implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -82,8 +81,17 @@ default String group() {
      * @since 1.8.0
      */
     default DynatraceApiVersion apiVersion() {
-        // if not specified, defaults to v1 for backwards compatibility.
-        return getEnum(this, DynatraceApiVersion.class, "apiVersion").orElse(DynatraceApiVersion.V1);
+        DynatraceApiVersion defaultVersion = DynatraceApiVersion.V2;
+        if (!deviceId().isEmpty()) {
+            // if deviceId is not empty, use v1 as default.
+            defaultVersion = DynatraceApiVersion.V1;
+        }
+
+        // If a device id is specified, use v1 as default. If it is not, use v2.
+        // The version can be overwritten explicitly when creating a MM config
+        // For Spring Boot, v1 is automatically chosen when the device id is set.
+        return getEnum(this, DynatraceApiVersion.class, "apiVersion")
+                .orElse(defaultVersion);
```

Comment:
What do you think about this (static imports `V1` and `V2`):
`.orElse(deviceId().isEmpty() ? V2 : V1);`